geopandas/tests/test_geom_methods.py

```diff
@@ -1791,3 +1791,44 @@ def test_force_3d(self):
             ],
         )
         assert_geoseries_equal(expected, self.g1.force_3d([1, 2]))
+
+    @pytest.mark.skipif(
+        shapely.geos_version < (3, 11, 0), reason="different order in GEOS<3.11"
+    )
+    def test_build_area(self):
+        s = GeoSeries.from_wkt(
+            [
+                "LINESTRING (18 4, 4 2, 2 9)",
+                "LINESTRING (18 4, 16 16)",
+                "LINESTRING (16 16, 8 19, 8 12, 2 9)",
+                "LINESTRING (8 6, 12 13, 15 8)",
+                "LINESTRING (8 6, 15 8)",
+                "LINESTRING (0 0, 0 3, 3 3, 3 0, 0 0)",
+                "POLYGON ((1 1, 2 2, 1 2, 1 1))",
+                "LINESTRING (10 7, 13 8, 12 10, 10 7)",
+            ],
+            crs=4326,
+        )
+
+        expected = GeoSeries.from_wkt(
+            [
+                "POLYGON ((0 3, 3 3, 3 0, 0 0, 0 3), (2 2, 1 2, 1 1, 2 2))",
+                "POLYGON ((13 8, 10 7, 12 10, 13 8))",
+                "POLYGON ((2 9, 8 12, 8 19, 16 16, 18 4, 4 2, 2 9), "
+                "(8 6, 15 8, 12 13, 8 6))",
+            ],
+            crs=4326,
+            name="polygons",
+        )
+        assert_geoseries_equal(expected, s.build_area(node=True))
+
+        expected = GeoSeries.from_wkt(
+            [
+                "POLYGON ((0 0, 0 3, 3 3, 3 0, 0 0))",
+                "POLYGON ((1 1, 1 2, 2 2, 1 1))",
+                "POLYGON ((10 7, 12 10, 13 8, 10 7))",
+            ],
+            crs=4326,
+            name="polygons",
+        )
```

Comment:
Ah, I think this might actually be the wrong expected result.

Doing the equivalent in shapely directly:

```
In [11]: polygons = shapely.from_wkt(            [
    ...:                 "LINESTRING (18 4, 4 2, 2 9)",
    ...:                 "LINESTRING (18 4, 16 16)",
    ...:                 "LINESTRING (16 16, 8 19, 8 12, 2 9)",
    ...:                 "LINESTRING (8 6, 12 13, 15 8)",
    ...:                 "LINESTRING (8 6, 15 8)",
    ...:                 "LINESTRING (0 0, 0 3, 3 3, 3 0, 0 0)",
    ...:                 "POLYGON ((1 1, 2 2, 1 2, 1 1))",
    ...:                 "LINESTRING (10 7, 13 8, 12 10, 10 7)",
    ...:             ])

In [12]: shapely.build_area(polygons)
Out[12]: 
array([<GEOMETRYCOLLECTION EMPTY>, <GEOMETRYCOLLECTION EMPTY>,
       <GEOMETRYCOLLECTION EMPTY>, <GEOMETRYCOLLECTION EMPTY>,
       <GEOMETRYCOLLECTION EMPTY>, <POLYGON ((0 0, 0 3, 3 3, 3 0, 0 0))>,
       <POLYGON ((1 1, 1 2, 2 2, 1 1))>,
       <POLYGON ((10 7, 12 10, 13 8, 10 7))>], dtype=object)
```

So this actually does happen element-wise. But the empty results get dropped in the explode step?

The result when first collecting in a GeometryCollection:

```
In [16]: shapely.build_area(shapely.GeometryCollection(list(polygons)))
Out[16]: <MULTIPOLYGON (((0 3, 3 3, 3 0, 0 0, 0 3), (2 2, 1 2, 1 1, 2 2)), ((13 8, 10...>

In [17]: list(shapely.build_area(shapely.GeometryCollection(list(polygons))).geoms)
Out[17]: 
[<POLYGON ((0 3, 3 3, 3 0, 0 0, 0 3), (2 2, 1 2, 1 1, 2 2))>,
 <POLYGON ((13 8, 10 7, 12 10, 13 8))>,
 <POLYGON ((2 9, 8 12, 8 19, 16 16, 18 4, 4 2, 2 9), (8 6, 15 8, 12 13, 8 6))>]
```

Reply:
When I looked at this when I reviewed, I thought this was to align with the same expectations  / logic as polygonize xref discussion in #2949 (but I don't really have a view on whether that is the "right" choice). 
```python
from geopandas import GeoSeries
s = GeoSeries.from_wkt(
            [
                "LINESTRING (18 4, 4 2, 2 9)",
                "LINESTRING (18 4, 16 16)",
                "LINESTRING (16 16, 8 19, 8 12, 2 9)",
                "LINESTRING (8 6, 12 13, 15 8)",
                "LINESTRING (8 6, 15 8)",
                "LINESTRING (0 0, 0 3, 3 3, 3 0, 0 0)",
                "POLYGON ((1 1, 2 2, 1 2, 1 1))",
                "LINESTRING (10 7, 13 8, 12 10, 10 7)",
            ],
            crs=4326,
        )
```

I think this is easier to see what's happening visually:
![image](https://github.com/geopandas/geopandas/assets/45483497/7616da33-f8f4-47f6-9e19-a22a4ef7bb9b)
Based on the docstring saying inner rings should become holes, I think this is at least consistent with that.

Reply:
@jorisvandenbossche is right. The key difference is whether we pass node=True or False, i.e. whether we call `union_all` or not. If False, we need to wrap geoms in a collection and this test is wrong.